conjure-java-core/src/main/java/com/palantir/conjure/java/util/JavaNameSanitizer.java

```diff
@@ -88,6 +90,11 @@ public String visitUnknown(String unknownType) {
         return value;
     }
 
+    /** Sanitizes parameter names for errors. */
+    public static String sanitizeErrorParameterName(String input) {
+        return SourceVersion.isName(input) && !RESERVED_ERROR_PARAMETER_NAMES.contains(input) ? input : escape(input);
```

Comment:
Don't think `SourceVersion.isName` is required. JavaPoet already does this check when[ we create a parameter.](https://github.com/palantir/javapoet/blob/c1ea09d451b7e82c13967e702a8dcb6684c06a44/javapoet/src/main/java/com/palantir/javapoet/ParameterSpec.java#L132-L145)

```suggestion
        return RESERVED_ERROR_PARAMETER_NAMES.contains(input) ? escape(input) : input;
```

Reply:
Doesn't it [throw](https://github.com/palantir/javapoet/blob/c1ea09d451b7e82c13967e702a8dcb6684c06a44/javapoet/src/main/java/com/palantir/javapoet/ParameterSpec.java#L143) if `SourceVersion.isName` is false?

Fwiw, I'm not sure that `escape(input)` would make it return true, but I mostly pattern-matched from the existing other sanitization logic.

Reply:
`escape` appends an underscore - I think it wouldn't make the variable valid, and this will end up throwing when we construct the parameter.

I think when this throws later, it'd be slightly confusing to see the escaped name instead of just the offending name.

Reply:
Isn't that also true of any of the other cases above? Or is there something I'm missing that makes this unique to this case? If not, I'd prefer to keep the code consistent, and follow-up with a PR to clean this up.

Reply:
It's true for the other cases above as well. Sure, happy to merge as-is.

Reply:
Sweet - will follow-up with another refactor